### PR TITLE
SEN-1022 -- Pull dynamoDbQueueSource into public module

### PR DESCRIPTION
**Pull request by notxt**
**Don't Merge Yet**

_Testing_
- verify npm publishing information https://www.npmjs.com/package/dynamodb-queue-source

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+var gulp = require('gulp');
+var gulpIstanbul = require('gulp-istanbul');
+var gulpMocha = require('gulp-mocha');
+
+gulp.task('test', function (cb) {
+  gulp.src(['dynamoDbQueueSource.js'])
+    .pipe(gulpIstanbul())
+    .pipe(gulpIstanbul.hookRequire())
+    .on('finish', function () {
+          gulp.src(['test/**/*.js'])
```
Comment (nackjicholson) [inline gulpfile.js:12]: 2sp indentation. It's probably like this in wherever you copied it from as well. :)

File: dynamoDbQueueSource.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+var _ = require('lodash');
+var AWS = require('aws-sdk');
+
+var dynamoDb = new AWS.DynamoDB();
+
+var dynamoDbQueueSource = { };
```
Comment (nackjicholson) [inline dynamoDbQueueSource.js]: gap.

File: dynamoDbQueueSource.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  );
+};
+
+dynamoDbQueueSource.scanToQueue = function(queue, options, callback) {
+  var scanTask = {
+    queue: queue,
+    options: options,
+    callback: callback,
+    hasMoreItems: true,
+    results: { itemCount: 0 }
```
Comment (nackjicholson) [inline dynamoDbQueueSource.js]: gap.

File: dynamoDbQueueSource.js (hunk 2 above)
Comment (nackjicholson) [inline dynamoDbQueueSource.js:8]: Lots of gaps in objects and arrays. I feel like we already settled this in our PHP, but it's back.

**Comment by nackjicholson** [comment]
You may want to make a quick readme with an example usage and a blurb. Hook it up to travis would be good as well. Feel free to make those separate tickets, I'm okay with the way this is.

**Comment by cberube** [comment]
@noTxt -- Could you make a ticket for this when you get a chance? -- I'd rather not have a pull without a ticket number (let's add that ticket to next sprint, since we probably won't quite get this wrapped up in the current sprint).

**Comment by notxt** [comment]
@cberube mos def

**Comment by nwarlen** [comment]
Quick comment on https://www.npmjs.com/package/dynamodb-queue-source. _Helps manages interactions with AWS DynamoDb_ should be _Helps **manage** interactions with AWS DynamoDb_

**Comment by nwarlen** [comment]
:ok:

**Comment by jinsyaoommen** [comment]
:ok:

**Comment by nackjicholson** [comment]
:ok:

**Pull request merged**